### PR TITLE
Schema and BCDB cleanup #157

### PR DESCRIPTION
**Pull request by BolaNasr**
clean schemaFactory class

### CODE REVIEW

File: Jumpscale/data/schema/SchemaFactory.py [hunk 1]
```diff
@@ -107,9 +118,9 @@ def _add(self, schema_text_path):
                     raise RuntimeError("should not be there")
                 else:
                     res.append(s)
-                    self.schemas[s.url]=s
+                    self.schemas[s.url] = s
 
-        if len(res)==0:
+        if len(res) == 0:
             j.shell()
```
Comment (andrewayoub) [inline Jumpscale/data/schema/SchemaFactory.py]: please remove this and all similar usage of j.shell()

File: Jumpscale/data/bcdb/BCDB.py [hunk 2]
```diff
@@ -93,11 +92,16 @@ def __init__(self, name=None, zdbclient=None):
...
         self.logger_enable()
-        self.logger.info("BCDB INIT DONE:%s"%self.name)
-
-    def redis_server_start(self,port=6380,secret="123456"):
+        self.logger.info("BCDB INIT DONE:%s" % self.name)
 
-        self.redis_server = RedisServer(bcdb=self,port=port,secret=secret)
+    def redis_server_start(self, port=6380, secret="123456"):
+        """
+        starts redis server that makes BCDB browser through RDM
```
Comment (zaibon) [inline Jumpscale/data/bcdb/BCDB.py]: didn't you mean `starts redis server that makes BCDB browsable through RDM` ?

File: Jumpscale/data/types/CollectionTypes.py [hunk 3]
```diff
@@ -44,6 +44,58 @@ class JSON(String):
...
+        Check whether provided value is a dict
+        """
+        return isinstance(value, dict)
+
+    def get_default(self):
+        return dict()
+
+    def clean(self,v):
+        if self.check(v):
+            return True
```
[the remark this answers is not in the document]
Comment (andrewayoub) [reply Jumpscale/data/types/CollectionTypes.py]: No I think clean method should return the value after being cleaned

[the remark this answers is not in the document]
Comment (BolaNasr) [reply Jumpscale/data/types/CollectionTypes.py]: Ok that's i mean if value correct return it after cleaned

[the remark this answers is not in the document]
Comment (ranatrk) [reply Jumpscale/data/types/CollectionTypes.py]: https://github.com/threefoldtech/jumpscale_core/commit/084feb3e8a7e8be16092b1f0be43dfa273836108

**Pull request merged**